File: d6/java/d6b.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.util.ArrayList;
import java.util.Arrays;

class Point {
    int x, y;
    Point(int x, int y) {
        this.x = x;
        this.y = y;
    }
}

class Input {
    Point[] points;
    int xmax, ymax, xdim, ydim;
    Input(ArrayList<Point> points, int xmax, int ymax) {
        this.points = new Point[points.size()];
        this.points = points.toArray(this.points);
        this.xmax = xmax;
        this.ymax = ymax;
        this.xdim = xmax + 1;
        this.ydim = ymax + 1;
    }
}
    
public class d6b {
    private static Input readInput(String fileName) {
        ArrayList<Point> points = new ArrayList<>();
        String line;
        int xmax = 0;
        int ymax = 0;

        try {
            BufferedReader br = new BufferedReader(new FileReader(fileName));
            while ((line = br.readLine()) != null) {
                String[] nums = line.split(",");
                if (nums.length != 2) throw new IllegalArgumentException("Too many numbers in line");
                int x = Integer.parseInt(nums[0].strip());
                int y = Integer.parseInt(nums[1].strip());
                if (x > xmax) xmax = x;
                if (y > ymax) ymax = y;
                if (x < 0 || y < 0 || x > Integer.MAX_VALUE - 1 || y > Integer.MAX_VALUE - 1) {
                    throw new IllegalArgumentException("Coordinates out of range");
                }
                points.add(new Point(x, y));
            }
            br.close();
        } catch (Exception e) {
            System.out.println(e.toString());
            System.exit(1);
        }
        return new Input(points, xmax, ymax);
    }

    private static int manhattan(Point p1, Point p2) {
        return Math.abs(p1.x - p2.x) + Math.abs(p1.y - p2.y);
    }

    private static int getAnswer(Input input, int maxdist) {
        Point[] points = input.points;
        int regions = 0;
        Point p2 = new Point(0, 0);

        for (int y = 0; y < input.ydim; y++) {
            for (int x = 0; x < input.xdim; x++) {
                int sumdist = 0;
                p2.x = x;
                p2.y = y;
                regions++;
                for (Point p : points) {
                    int dist = manhattan(p, p2);
                    sumdist += dist;
                    if (sumdist >= maxdist) {
                        regions--;
                        break;
                    }
                }
            }
        }
        return regions;
    }

    public static void main(String[] args) {
        int answer = 0;
        final Input input = readInput("input.txt");
        long start = System.nanoTime();
        answer = getAnswer(input, 10000);
        long end = System.nanoTime() - start;
        System.out.println("answer :" + answer);
        System.out.println("timing :" + end / 1000000 + " ms");
    }
}

File: d6/py/d6b.py
import re
from time import perf_counter_ns

class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

class Info:
    def __init__(self, points, xmax, ymax):
        self.points = points
        self.xmax = xmax
        self.ymax = ymax
        self.xdim = xmax + 1
        self.ydim = ymax + 1

    def onborder(self, p):
        return p.x == 0 or p.y == 0 or p.x == self.xmax or p.y == self.ymax

def readInput(fileName):
    points = []
    xmax = ymax = 0
    p = re.compile(r'(\d+),\s(\d+)')

    try:
        info = open(fileName)
        for line in info:
            m = p.match(line)
            if m == None:
                raise Exception(fileName + " is wrong input file")
            x = int(m.group(1))
            y = int(m.group(2))
            if x < 0 or y < 0:
                raise Exception(fileName + " is wrong input file")
            if x > xmax:
                xmax = x
            if y > ymax:
                ymax = y
            points.append(Point(x, y))

    except IOError as ioerror:
        print("{} {}".format(ioerror.filename, ioerror.strerror))
        exit()
    except Exception as error:
        print(error.args)
        exit()

    return Info(points, xmax, ymax)

def manhattan(p1, p2):
    return abs(p1.x - p2.x) + abs(p1.y - p2.y)

def getAnswer(input, maxdist):
    points = input.points
    p2 = Point(0, 0)
    regions = 0

    for y in range(input.ydim):
        for x in range(input.xdim):
            p2.x = x
            p2.y = y
            regions += 1
            sumdist = 0
            for p in points:
                sumdist += manhattan(p, p2)
                if sumdist >= maxdist:
                    regions -= 1
                    break
    return regions

def main():
    answer = 0
    input = readInput("input.txt")
    timing = perf_counter_ns()
    answer = getAnswer(input, 10000)
    timing = perf_counter_ns() - timing
    print("Answer: {}".format(answer))
    print("timing: {} ms".format(timing // 1000000))

if __name__ == "__main__": main()